Plugin/CustomerPlugin.php

```diff
@@ -0,0 +1,230 @@
+        \Magento\Reward\Helper\Data $rewardHelper,
+        \Dotdigitalgroup\Enterprise\Helper\Data $helper
+    ) {
+        $this->dateTime = $dateTime;
+        $this->rewardHistoryCollectionFactory = $rewardHistoryCollectionFactory;
+        $this->customerSegmentCustomerResource = $customerSegmentCustomerResource;
+        $this->rewardHelper = $rewardHelper;
+        $this->helper = $helper;
+    }
+
+    /**
+     * @param \Dotdigitalgroup\Email\Model\Apiconnector\Customer $subject
+     * @param \Magento\Customer\Model\Customer $customer
+     * @return mixed
+     */
+    public function beforeSetCustomerData(\Dotdigitalgroup\Email\Model\Apiconnector\Customer $subject, $customer)
+    {
+        $this->customer = $customer;
+        $websiteId = $customer->getWebsiteId();
+        $this->reward = false;
+
+        if ($this->helper->getRewardPointMapping($websiteId)) {
+            $customer->setRewardPoints($this->getRewardPoints());
+        }
+        if ($this->helper->getRewardAmountMapping($websiteId)) {
+            $customer->setRewardAmmount($this->getRewardAmmount());
+        }
+        if ($this->helper->getExpirationDateMapping($websiteId)) {
+            $customer->setExpirationDate($this->getExpirationDate());
+        }
+        if ($this->helper->getLastUsedDateMapping($websiteId)) {
+            $customer->setLastUsedDate($this->getLastUsedDate());
+        }
+        if ($this->helper->getCustomerSegmentMapping($websiteId)) {
+            $customer->setCustomerSegments($this->getCustomerSegments());
+        }
+
+        return [$customer];
+    }
+
+    /**
+     * Reward points balance.
+     *
+     * @return int
+     */
+    public function getRewardPoints()
+    {
+        if (! $this->reward) {
+            $this->_setReward();
+        }
+
+        if ($this->reward !== true) {
+            return $this->reward->getPointsBalance();
+        }
+
+        return '';
+    }
+
+    /**
+     * Currency amount points.
+     *
+     * @return mixed
+     */
+    public function getRewardAmmount()
+    {
+        if (!$this->reward) {
+            $this->_setReward();
+        }
+
+        if ($this->reward !== true) {
+            return $this->reward->getCurrencyAmount();
+        }
+
+        return '';
+    }
+
+    /**
+     * Expiration date to use the points.
+     *
+     * @return string
+     */
+    public function getExpirationDate()
+    {
+        //set reward for later use
+        if (!$this->reward) {
+            $this->_setReward();
+        }
+
+        if ($this->reward !== true) {
+            $expiredAt = $this->reward->getExpirationDate();
+
+            if ($expiredAt) {
+                $date = $this->dateTime->formatDate($expiredAt, true);
+            } else {
+                $date = '';
+            }
+
+            return $date;
+        }
+
+        return '';
+    }
+
+    /**
+     * Get the customer reward.
+     *
+     * @return void
+     */
+    public function _setReward()
```

Comment:
@adeelq this is a public function why using _underscore _setReward ?

Reply:
@cdiacon not a new code. just copied it.

Reply:
remove underscore